src/contribute/precommit-triage-guide.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: section
+title: "Precommit Slowness Triage Guide"
+permalink: /contribute/precommit-triage-guide/
+section_menu: section-menu/contribute.html
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Precommit Slowness Triage Guide
+
+Beam precommit jobs are suites of tests run automatically on Jenkins build
+machines for each pull request (PR) submitted to
+[apache/beam](https://github.com/apache/beam). For more information and the
+difference between precommits and postcommits, see
+[testing](/contribute/testing/).
+
+## What are fast precommits?
+
+Precommit tests are required to pass before a pull request (PR) may be merged.
+When these tests are slow they slow down Beam's development process.
+
+The aim is to have 95% of precommit jobs complete within 30 minutes
+(failing or passing).
+Technically, the 95th percentile of running time should be below 30 minutes over
+the past 4 weeks, where running time is the duration of time the job spends in
+the Jenkins queue + the actual time it spends running.
+
+## Detemining Slowness
```

Comment:
See comment below about timeouts as possible solution. Another option I'm considering would be to return a special error result from the precommit run in case it goes over 30m. This way the job isn't aborted so there's less frustration, but the author and reviewer get a reminder that precommits are slow.

Reply:
I'll probably go for Jenkins timeouts.